app/components/data/details/download-file-group-dialog/download-file-group-dialog.component.ts

```diff
@@ -53,12 +55,21 @@ export class DownloadFileGroupDialogComponent {
             const subject = new AsyncSubject();
             task.progress.next(1);
             this._getListOfFilesToDownload().subscribe((files) => {
-                task.progress.next(10);
-                const downloadObs = this._downloadFiles(task, folder, files);
-                Observable.combineLatest(downloadObs).subscribe(() => {
-                    this.shell.showItemInFolder(folder);
+                if (files.count() === 0) {
+                    this.notificationService.warn(
+                        "Pattern not found",
+                        `Failed to find pattern: ${this._getPatterns()}`,
+                    );
+                    task.progress.next(100);
                     subject.complete();
-                });
+                } else {
+                    task.progress.next(10);
+                    const downloadObs = this._downloadFiles(task, folder, files);
+                    Observable.combineLatest(downloadObs).subscribe(() => {
```

Comment:
you don't need combineLatest here

Reply:
The downloadObs is collection of Observables. I think we do need combineLatest.